packages/teleport/src/components/LabelSelector/LabelSelector.tsx

```diff
@@ -0,0 +1,269 @@
+    }
+  }, [showAdd]);
+
+  useClickOutside(addLabelContainerRef, () => {
+    setShowAdd(false);
+  });
+
+  useEscape(() => {
+    setShowAdd(false);
+  });
+
+  const handleAddLabel = () => {
+    setLabels([...labels, newLabel.trim()]);
+    setNewLabel('');
+  };
+
+  const handleLabelDismiss = (label: string) => {
+    const labelList = [...labels];
+    labelList.splice(labelList.indexOf(label), 1);
+    setLabels(labelList);
+  };
+
+  return (
+    <div>
+      <Heading>
+        <Text style={{ float: 'left' }}>Assign Labels (optional)</Text>
+        <div ref={infoIconRef} style={{ marginLeft: '12px', float: 'left' }}>
+          <Info
+            style={{
+              cursor: 'pointer',
+              fontSize: '16px',
+              paddingTop: '5px',
+            }}
+            onClick={() => setShowTooltip(!showTooltip)}
+          />
+        </div>
+        <Popover
+          id="simple-popper"
+          open={showTooltip}
+          anchorOrigin={{
+            vertical: 'bottom',
+            horizontal: 'right',
+          }}
+          transformOrigin={{
+            vertical: 'top',
+            horizontal: 'left',
+          }}
+          anchorEl={infoIconRef.current}
+          onClose={() => setShowTooltip(false)}
+        >
+          <Box
+            bg="#011223"
+            color="white"
+            width={362}
+            p={4}
+            style={{
+              boxShadow: '0px 8px 14px rgba(12, 12, 14, 0.07)',
+              borderRadius: '8px',
+            }}
+          >
+            Teleport provides users the ability to add labels (in the form of
+            key:value pairs) to resources. Some valid example labels are “env:
+            prod” and “arch: x86_64”. Labels, used in conjunction with roles,
+            define access in Teleport. For example, you can specify that users
+            with the “on-call” role can access resources labeled “env: prod”.
+            For more information, check out our documentation on{' '}
+            <a
+              href="https://goteleport.com/docs/setup/admin/trustedclusters/"
+              target="_blank"
+            >
+              RBAC
+            </a>{' '}
+            and{' '}
+            <a
+              href="https://goteleport.com/docs/setup/admin/labels/"
+              target="_blank"
+            >
+              labels
+            </a>
+            .
+          </Box>
+        </Popover>
+        <Text style={{ float: 'right' }}>
+          <a
+            href="https://goteleport.com/docs/setup/admin/labels/"
+            target="_blank"
+            style={{ color: 'rgb(255, 255, 255)' }}
+          >
+            View Documentation
+          </a>
+        </Text>
+      </Heading>
+      <LabelContainer
+        onClick={() => setShowAdd(!showAdd)}
+        data-testid="label-container"
+      >
+        {labels.length === 0 && (
+          <Text style={{ color: 'rgba(255, 255, 255, 0.1)' }}>
```

Comment:
`Text` comp allows you to pass in `color` using styled system

Reply:
updated